src/Cardano/Prelude/GHC/Heap.hs

```diff
@@ -0,0 +1,50 @@
+{-# LANGUAGE DoAndIfThenElse #-}
+
+-- | This code has been adapted from the module 'GHC.AssertNF' of the
+--   package 'ghc-heap-view' authored by Joachim Breitner.
```

Comment:
This is probably good enough, but I wonder if referencing a file / commit from the `ghc-heap-view` would make things even more clear. 

`ghc-heap-view` [has a BSD3 license](http://hackage.haskell.org/package/ghc-heap-view), and this [SO answer](https://softwareengineering.stackexchange.com/questions/234437/bsd-mit-license-changes-when-copy-pasting-a-function-from-another-bsd-mit-licens) seems to indicate citing the source is good enough. 

IANAL, though.

Reply:
I would add a link to the hackage page and the github repo. I don't think referencing a file or git commit is necessary.